telegram/bot.py

```diff
@@ -1573,6 +1583,32 @@ def answer_inline_query(self,
         """
         url = '{}/answerInlineQuery'.format(self.base_url)
 
+        if current_offset is not None and next_offset is not None:
+            raise ValueError('`current_offset` and `next_offset` are mutually exclusive!')
+
+        if current_offset is not None:
+            if current_offset == '':
+                current_offset = 0
+            else:
+                current_offset = int(current_offset)
+
+            next_offset = ''
+
+            if callable(results):
+                results = results(current_offset)
+                if not results:
+                    results = []
+                else:
+                    next_offset = current_offset + 1
```

Comment:
I don't understand why there is an `else` block here.
- [ ] Explain

Reply:
```python
# callback returned None, i.e. there are no more results. We still need to answer the query to tell TG that.
# so we pass an empty results list along with next_offset = '', which tells TG to not ask again
if not results:    
    results = []
# callback returned more results, so we need to assume that there are results for the next page as well
# hence, increase next_offset. If there are no more results fo the next page, the next query will trigger 
# the `if not results` clause above
else:
    next_offset = current_offset + 1
```